linkerd/cache/src/lib.rs

```diff
@@ -79,7 +79,7 @@ where
                     // Drop defunct services before interacting with the cache.
                     let n = services.len();
                     services.retain(|_, (_, weak)| {
-                        if weak.upgrade().is_some() {
+                        if weak.strong_count() > 0 {
```

Comment:
:+1:

I think some of the performance advantage of this change is also that we don't increment the strong count and then immediately decrement it again (when dropping the returned `Arc` from `upgrade`).